frontend/data.ts

```diff
@@ -1,101 +1,99 @@
 import Replicache, { ReadTransaction, WriteTransaction } from "replicache";
 import { useSubscribe } from "replicache-react-util";
-import { getShape, Shape } from "../shared/shape";
-import { getClientState } from "../shared/client-state";
-import {
-  createShape,
-  CreateShapeArgs,
-  moveShape,
-  MoveShapeArgs,
-  overShape,
-  OverShapeArgs,
-} from "../shared/mutators";
+import { getShape, Shape, putShape, moveShape } from "../shared/shape";
+import { getClientState, overShape } from "../shared/client-state";
 import type Storage from "../shared/storage";
 import { newID } from "../shared/id";
 
 /**
  * Abstracts Replicache storage (key/value pairs) to entities (Shape).
  */
-export class Data {
-  private rep: Replicache;
-  readonly clientID: string;
+export type Data = ReturnType<typeof createData>;
 
-  constructor(rep: Replicache) {
-    this.rep = rep;
+export function createData(rep: Replicache) {
+  // TODO: Use clientID from Replicache:
+  // https://github.com/rocicorp/replicache-sdk-js/issues/275
+  let clientID = localStorage.clientID;
```

Comment:
localStorage is slow. Is this called a lot?

Reply:
No, it's just called once on page load.

Also, please expose Replicache's client ID then I wouldn't need this :). Also the user chip wouldn't flicker on startup.

https://github.com/rocicorp/replicache-sdk-js/issues/275